libmamba/src/core/mirror.cpp

```diff
@@ -0,0 +1,444 @@
+// Copyright (c) 2023, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include <fmt/format.h>
+
+#include "mamba/core/mirror.hpp"
+#include "mamba/core/output.hpp"
+#include "mamba/util/string.hpp"
+#include "mamba/util/url_manip.hpp"
+
+#include "curl.hpp"
+
+namespace mamba
+{
+
+    /*************************
+     * Mirror implementation *
+     *************************/
+
+    Mirror::Mirror(const std::string& url)
```

Comment:
I would prefer as many as possible members to be initialized in the class declaration, to help reduce noise in functions. Here mostly the initializations through constant which type  dont necessitate to add includes in the header.